bots/run-queue

```diff
@@ -56,41 +113,32 @@ def main():
     channel = connection.channel()
 
     try:
-        declare_public_result = channel.queue_declare(queue='public', passive=True, auto_delete=False)
-        declare_rhel_result = channel.queue_declare(queue='rhel', passive=True, auto_delete=False)
+        declare_webhook_result = channel.queue_declare(queue='webhook', passive=True, auto_delete=False)
     except pika.exceptions.ChannelClosed:
-        print("One of the queues doesn't exist")
         return 1
 
-    queue='public'
-    if redhat_network():
-        # Try the rhel queue if the public queue is empty
-        if declare_public_result.method.message_count == 0:
-            queue = 'rhel'
-        # If both are non-empty, shuffle
-        elif declare_rhel_result.method.message_count > 0:
-            queue = ['public', 'rhel'][random.randrange(2)]
+    arguments = {
+        "x-max-priority": MAX_PRIORITY
+    }
+    declare_public_result = channel.queue_declare(queue='public', arguments=arguments, auto_delete=False)
+    declare_rhel_result = channel.queue_declare(queue='rhel', arguments=arguments, auto_delete=False)
 
-    # Get one item from the queue if present
-    method_frame, header_frame, body = channel.basic_get(queue=queue)
-    if method_frame:
-        ret = 0
-        body = json.loads(body)
-        sys.stderr.write("Consuming {0} task:\n{1}\n".format(body['type'],json.dumps(body, indent=2, sort_keys=True)))
-        sys.stderr.flush()
-        p = subprocess.Popen(body['command'], shell=True)
-        while p.poll() is None:
-            connection.process_data_events()
-            time.sleep(3)
-
-        if p.returncode in [0, 2]:
-            channel.basic_ack(method_frame.delivery_tag)
-    else:
-        # nothing in queue, no work done
-        ret = 1
+
+    cmd, delivery_tag = consume_webhook_queue(channel, opts.amqp)
+    if not cmd:
+        cmd, delivery_tag = consume_task_queue(channel, opts.amqp, declare_public_result, declare_rhel_result)
+    assert cmd, logging.warning("All queues are empty")
```

Comment:
Heh, this is pretty garbled :) (drop the assert). As said above, let's use logging.info here, as this is a normal state. return if cmd is false-y.